packages/data/src/components/with-dispatch/index.js

```diff
@@ -79,61 +73,17 @@ import { RegistryConsumer } from '../registry-provider';
  * @return {Component} Enhanced component with merged dispatcher props.
  */
 const withDispatch = ( mapDispatchToProps ) => createHigherOrderComponent(
-	( WrappedComponent ) => {
-		class ComponentWithDispatch extends Component {
-			constructor( props ) {
-				super( ...arguments );
-
-				this.proxyProps = {};
-
-				this.setProxyProps( props );
-			}
-
-			proxyDispatch( propName, ...args ) {
-				// Original dispatcher is a pre-bound (dispatching) action creator.
-				mapDispatchToProps( this.props.registry.dispatch, this.props.ownProps, this.props.registry )[ propName ]( ...args );
-			}
-
-			setProxyProps( props ) {
-				// Assign as instance property so that in subsequent render
-				// reconciliation, the prop values are referentially equal.
-				// Importantly, note that while `mapDispatchToProps` is
-				// called, it is done only to determine the keys for which
-				// proxy functions should be created. The actual registry
-				// dispatch does not occur until the function is called.
-				const propsToDispatchers = mapDispatchToProps( this.props.registry.dispatch, props.ownProps, this.props.registry );
-				this.proxyProps = mapValues( propsToDispatchers, ( dispatcher, propName ) => {
-					if ( typeof dispatcher !== 'function' ) {
-						// eslint-disable-next-line no-console
-						console.warn( `Property ${ propName } returned from mapDispatchToProps in withDispatch must be a function.` );
-					}
-					// Prebind with prop name so we have reference to the original
-					// dispatcher to invoke. Track between re-renders to avoid
-					// creating new function references every render.
-					if ( this.proxyProps.hasOwnProperty( propName ) ) {
-						return this.proxyProps[ propName ];
-					}
-
-					return this.proxyDispatch.bind( this, propName );
-				} );
-			}
-
-			render() {
-				return <WrappedComponent { ...this.props.ownProps } { ...this.proxyProps } />;
-			}
+	( WrappedComponent ) => pure(
```

Comment:
I'm pretty sure this `pure` call is not really necessary (and I suspect the small decline of performance that you noticed is coming from it)

Reply:
oh good point, I'll try without it.

Reply:
Removed in 6500117.  The new 3 run average for the new `withDispatch` after `pure` being removed is:

Average time to load: 6590ms
Average time to DOM content load: 6041ms
Average time to type character: 99ms
Slowest time to type character: 234ms
Fastest time to type character: 71ms

So ya, it does look like there was some unnecessary overhead added by using `pure` here.